sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcSourceSuite.scala

```diff
@@ -160,6 +160,16 @@ abstract class OrcSuite extends OrcTest with BeforeAndAfterAll {
       }
     }
   }
+
+  // This is a test case for ORC-285
+  test("SPARK-23340 Empty float/double array columns raise EOFException") {
+    Seq(Seq(Array.empty[Float]).toDF(), Seq(Array.empty[Double]).toDF()).foreach { df =>
```

Comment:
Since this is the issue when we use vectorized ORC reader, please check both vectorized and non-vectorized readers.

Reply:
uh, I see. I just saw [your reply to omalley](https://github.com/apache/spark/pull/20511#discussion_r167952000)